From 95028675e877d7c4b2604f13113ac6b552f8b902 Mon Sep 17 00:00:00 2001
From: Chun-wei Fan <fanchunwei@src.gnome.org>
Date: Fri, 18 Sep 2020 15:06:36 +0800
Subject: [PATCH] gicsanner/message.py: Windows: Fix running on different
 drives

os.path.relpath() will throw a ValueError when the two paths that
are fed to it are on different drives, at least when running under
cmd.exe consoles.

Fix this by falling back to the full path when this ValueError is
thrown.
---
 giscanner/message.py | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/giscanner/message.py b/giscanner/message.py
index 9057c23c1..24f1b7dfe 100644
--- a/giscanner/message.py
+++ b/giscanner/message.py
@@ -75,8 +75,14 @@ class Position(object):
                                         self.column or -1)
 
     def format(self, cwd):
-        filename = os.path.relpath(os.path.realpath(self.filename),
-                                   os.path.realpath(cwd))
+        # Windows: We may be using different drives self.filename and cwd,
+        #          which leads to a ValuError when using os.path.relpath().
+        #          In that case, just use the entire path of self.filename
+        try:
+            filename = os.path.relpath(os.path.realpath(self.filename),
+                                       os.path.realpath(cwd))
+        except ValueError:
+            filename = os.path.realpath(self.filename)
 
         if self.column is not None:
             return '%s:%d:%d' % (filename, self.line, self.column)
-- 
GitLab

